syntax = "proto3";

// to compile this .proto file and generate C++ output, put it in a folder with protoc.exe (found in: 3rdparty/protobuf/bin)
// then open cmd and enter the following commands:
// cd <your folder path>
// protoc -I=. --cpp_out=. framedata.proto

package proto;

message ColorImage {
	bytes data = 1;
	int32 width = 2;
	int32 height = 3;
	uint64 size = 4;
	uint64 timestamp = 5;
	int32 stridebytes = 6;
	string format = 7;
}

message DepthImage {
	bytes data = 1;
	int32 width = 2;
	int32 height = 3;
	uint64 size = 4;
	uint64 timestamp = 5;
	int32 stridebytes = 6;
	string format = 7;
}

message FrameData {
	ColorImage colorImage = 1;
	DepthImage depthImage = 2;
}

message Intrinsics {
	int32 width = 1;
	int32 height = 2;
	double cx = 3;
	double cy = 4;
	double fx = 5;
	double fy = 6;
	double k1 = 7;
	double k2 = 8;
	double k3 = 9;
	double k4 = 10;
	double k5 = 11;
	double k6 = 12;
	double codx = 13;
	double cody = 14;
	double p1 = 15;
	double p2 = 16;
	double metric_radius = 17;
}

message Calibration {
	string data = 1;
	uint32 length = 2;
}

message Data {
	bytes data = 1;
	uint64 size = 2;
}